### PR TITLE
Python 3 port of source.

### PR DESCRIPTION
**Pull request by codito**
Update source to work with both py2 and py3.
Add tox for running tests together.
Limit sandhi tests to reference file to make them pass (and used it as a reference for python port).

### CODE REVIEW

File: tests/test_sandhi.py [hunk 1]
```diff
@@ -39,11 +40,37 @@ def test_sandhi_split(sandhiobj, sandhi_reference):
     assert sandhi_reference[0] in splits
 
 def load_reference_data():
+    files = ['refs.txt',
```
Comment (codito) [inline tests/test_sandhi.py]: Made this change to ensure we have some tests passing to validate functionality. We can uncomment the files one by one and fix the failing test cases.

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,111 @@
+
+# Created by https://www.gitignore.io/api/python
```
Comment (codito) [inline .gitignore]: Included this generated gitignore to make it easier to work in virtualenv.

**Comment by kmadathil** [comment]
Looks good to me. I've requested a review from @avinashvarna as well. I have merged locally and it seems fine. I'll push once I've heard back from Avinash.

I'll add @codito to the contributors list so you can work on the main repo!

File: tests/test_sandhi.py (hunk 1 above)
Comment (kmadathil) [reply tests/test_sandhi.py]: It'd be good to have a way to extract the passing/failing tests from the UoHD files so we can add the passing ones to our regression set, and have an easy-to-work-on list of failing tests. @avinashvarna  was working on sandhi, but is currently occupied, I think.  

Something similar needs to be done on the LexicalAnalyzer tests as well. I've been off working on the Morphological Analyzer, and have neglected this.

**Pull request merged**